doc/source/methods/modeldistillation.ipynb

```diff
@@ -0,0 +1,180 @@
+    "\n",
+    "ad = AdversarialAE(\n",
+    "    distilled_model=distilled_model,\n",
+    "    model=model,\n",
+    "    temperature=0.5\n",
+    ")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Fit\n",
+    "\n",
+    "We then need to train the detector. The following parameters can be specified:\n",
+    "\n",
+    "* `X`: training batch as a numpy array.\n",
+    "\n",
+    "* `loss_fn`: loss function used for training. Defaults to the custom adversarial loss.\n",
+    "\n",
+    "* `optimizer`: optimizer used for training. Defaults to [Adam](https://arxiv.org/abs/1412.6980) with learning rate 1e-3.\n",
+    "\n",
+    "* `epochs`: number of training epochs.\n",
+    "\n",
+    "* `batch_size`: batch size used during training.\n",
+    "\n",
+    "* `verbose`: boolean whether to print training progress.\n",
+    "\n",
+    "* `log_metric`: additional metrics whose progress will be displayed if verbose equals True.\n",
+    "\n",
+    "* `preprocess_fn`: optional data preprocessing function applied per batch during training.\n",
+    "\n",
+    "\n",
+    "```python\n",
+    "ad.fit(X_train, epochs=50)\n",
+    "```\n",
+    "\n",
+    "The threshold for the adversarial / harmfulness score can be set via ```infer_threshold```. We need to pass a batch of instances $X$ and specify what percentage of those we consider to be normal via `threshold_perc`. Even if we only have normal instances in the batch, it might be best to set the threshold value a bit lower (e.g. $95$%) since the the model could have misclassified training instances leading to a higher score if the reconstruction picked up features from the correct class or some instances might look adversarial in the first place.\n",
+    "\n",
+    "```python\n",
+    "ad.infer_threshold(X_train, threshold_perc=95, batch_size=64)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Detect\n",
+    "\n",
+    "We detect adversarial / harmful instances by simply calling `predict` on a batch of instances `X`. We can also return the instance level score by setting `return_instance_score` to True.\n",
+    "\n",
+    "The prediction takes the form of a dictionary with `meta` and `data` keys. `meta` contains the detector's metadata while `data` is also a dictionary which contains the actual predictions stored in the following keys:\n",
+    "\n",
+    "* `is_adversarial`: boolean whether instances are above the threshold and therefore adversarial instances. The array is of shape *(batch size,)*.\n",
+    "\n",
+    "* `instance_score`: contains instance level scores if `return_instance_score` equals True.\n",
+    "\n",
+    "\n",
+    "```python\n",
+    "preds_detect = ad.predict(X, batch_size=64, return_instance_score=True)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Examples\n",
+    "\n",
+    "### Image\n",
```

Comment:
Maybe change to "Image drift detection" b/c it is not an adversarial detection example.

Reply:
Harmful drift detection through model distillation on CIFAR10